### PR TITLE
WIP Draft of a hidden doc about the experimental Translator

### PR DESCRIPTION
**Pull request by ndintenfass**

### CODE REVIEW

File: jekyll/_cci2/migrating-from-1-2-experimental-translator.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+{:toc}
+
+## Important Caveat:
+This Translator is designed to produce working configuration, but not necessarily optimal configuration. The configuration generated is commented to help you understand the inferred commands that the 1.0 build engine executed. _For production use, we strongly recommend reading and manually creating your 2.0 configuration using our [documentation on migrating from 1.0 to 2.0]({{ site.baseurl }}/2.0/migrating-from-1-2/)_.
+
+## Translator Overview
+The configuration Translator is an experimental project to help with migrating from CircleCI 1.0 to 2.0. 
+
+The approach in CircleCI 2.0 differs in a few significant ways that make an automated migration complicated:
+1. **CircleCI 1.0 has "magic" inferred commands, whereas 2.0 is wholly deterministic.** In CircleCI 1.0 our Inference Engine runs implicit commands based on the contents of your repository in many cases. For instance, unless overridden, CircleCI 1.0 would detect the presence of rspec tests and run them automatically with no intervention. In 2.0 our philosophy is to run no "magic" commands, preferring instead to have all configuration be transparent and deterministic for developers.
```
Comment (marcomorain) [inline jekyll/_cci2/migrating-from-1-2-experimental-translator.md]: I would personally use the word `explicit` over `transparent` in this document, I think it better conveys the reasoning behind the verbosity of 2.0 config files.

File: jekyll/_cci2/migrating-from-1-2-experimental-translator.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+1. **CircleCI 1.0 has "magic" inferred commands, whereas 2.0 is wholly deterministic.** In CircleCI 1.0 our Inference Engine runs implicit commands based on the contents of your repository in many cases. For instance, unless overridden, CircleCI 1.0 would detect the presence of rspec tests and run them automatically with no intervention. In 2.0 our philosophy is to run no "magic" commands, preferring instead to have all configuration be transparent and deterministic for developers. 
+2. **CircleCI 1.0 has an opinionated lifecycle, whereas 2.0 is built around arbitrary jobs orchestrated with workflows.** In CircleCI 1.0 your build consists of specific stages that are baked into the system, and the configuration semantics are structured around those stages. In CircleCI 2.0 we allow for arbitrary jobs that are orchestrated with Workflows, so the idiom for how to structure the various tasks of your build is quite different and more powerful in 2.0.
+
+## Translator Limitations
+### Supported Languages
+The Translator currently attempts to support transfer of inferred commands for the following languages (with some limitations for each):
+* Ruby
+* PHP
+* Node.js
+* iOS (partial)
```
Comment (marcomorain) [inline jekyll/_cci2/migrating-from-1-2-experimental-translator.md]: Let's call out here that "partial" refers to no automatic code-signing.

`* iOS (partial - 1.0 code signing is not supported)`

File: jekyll/_cci2/migrating-from-1-2-experimental-translator.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+The API 1.1 endpoint is:
+
+`GET: /project/:vcs-type/:username/:project/config-translation`
+
+For instance, to use the Translator from your browser (assuming you are authenticated in circleci.com) for a repository called `foo` in a GitHub org named `bar` you could request this URL in your browser:
+
+`https://circleci.com/api/v1.1/project/github/bar/foo/config-translation`
+
+You can also optionally pass your `circle-token` directly and/or the `branch` you want to translate. For instance, if you were to call this with `curl` for the above theoerical project (assuming you had your CircleCI API token in an environment variable called `CIRCLE_TOKEN` it might look like:
```
Comment (marcomorain) [inline jekyll/_cci2/migrating-from-1-2-experimental-translator.md]: for private repos, the token is required so `optionally pass your 'circle-token'` doesn't sound correct to me here. Maybe just `You can also optionally pass the 'branch' you want to translate`?

**Comment by marcomorain** [comment]
This is awesome, thanks @ndintenfass

File: jekyll/_cci2/migrating-from-1-2-experimental-translator.md (hunk 3 above)
Comment (ndintenfass) [reply jekyll/_cci2/migrating-from-1-2-experimental-translator.md]: Optional in that if you’re logged in and hit the endpoint in your browser you shouldn’t need a token. Will clarify.

Comment (marcomorain) [reply jekyll/_cci2/migrating-from-1-2-experimental-translator.md]: :+1:

**Comment by ndintenfass** [comment]
Superseded by https://github.com/circleci/circleci-docs/pull/1728

**Pull request closed without merging**